### DISCUSSION TITLE
Is it possible to know whether a page is using SSR or SSG in _app.getInitialProps?

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/next.js#12216] [Help]
In `_app.getInitialProps`, it'd be useful to know wether the page is using SSR or SSG, in order to perform some automated processing.

One example would be to bypass most of the work related to SSR pages, when building SSG pages. See https://github.com/zeit/next.js/discussions/10909#discussioncomment-1247

I'm thinking the `props` provided to `getInitialProps` could contain such information.

**Comment by andreas1327250** [comment]
I solved it by setting a boolean flag on the page itself (like you would define getInitialProps)

MyPage.hasStaticProps = true

**Reply by Vadorequest** [reply]
It's similar to https://github.com/zeit/next.js/discussions/10949#discussioncomment-6148

I'm not a fan, it's easy to make a mistake when removing SSG from a page (keep the boolean on while using SSR), or forgetting to add it. 

I'm looking for a more robust way of doing things, automated.
